internal/api/download.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+)
+
+func (c *ApiClient) Download(ctx Context, reportType string, reportJournalType string, reportScheduleType string, reportAccountType string, reportDebtType string, dateOfTransaction string, dateFrom string, dateTo string, email string) error {
+	var body bytes.Buffer
+	var dateTransformed *model.Date
+	var toDateTransformed *model.Date
+	var fromDateTransformed *model.Date
+
+	if dateOfTransaction != "" {
+		raisedDateFormatted := model.NewDate(dateOfTransaction)
+		dateTransformed = &raisedDateFormatted
+	}
+
+	if dateTo != "" {
+		startDateFormatted := model.NewDate(dateTo)
+		toDateTransformed = &startDateFormatted
+	}
+
+	if dateFrom != "" {
+		endDateFormatted := model.NewDate(dateFrom)
+		fromDateTransformed = &endDateFormatted
+	}
+
+	err := json.NewEncoder(&body).Encode(model.Download{
+		ReportType:         reportType,
+		ReportJournalType:  reportJournalType,
+		ReportScheduleType: reportScheduleType,
+		ReportAccountType:  reportAccountType,
+		ReportDebtType:     reportDebtType,
+		DateField:          dateTransformed,
+		ToDateField:        toDateTransformed,
+		FromDateField:      fromDateTransformed,
+		Email:              email,
+	})
+	if err != nil {
+		return err
+	}
+
+	req, err := c.newBackendRequest(ctx, http.MethodGet, "/downloads", &body)
+
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusCreated {
+		return nil
+	}
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return ErrUnauthorized
+	}
+
+	if resp.StatusCode == http.StatusUnprocessableEntity {
+		var v model.ValidationError
+		if err := json.NewDecoder(resp.Body).Decode(&v); err == nil && len(v.Errors) > 0 {
+			return model.ValidationError{Errors: v.Errors}
+		}
+	}
+
+	if resp.StatusCode == http.StatusBadRequest {
+		var badRequests model.BadRequests
+		if err := json.NewDecoder(resp.Body).Decode(&badRequests); err != nil {
+			return err
+		}
+
+		validationErrors := make(model.ValidationErrors)
```

Comment:
(this may be a personal opinion) but I prefer to do `model.ValidationErrors{}` and reserve `make` for only when you want a certain size/capacity